docs/source/notes/serialization.rst

```diff
@@ -2,6 +2,59 @@
 Serialization semantics
 =======================
 
+Storage sharing is preserved in serialization
+---------------------------------------------
+
+.. _preserve-storage-sharing:
+
+PyTorch saves the underlying storages so that tensors sharing the same storage before :func:`torch.save`
+will still share storage after :func:`torch.load`.
+
+::
+
+    >>> tensor = torch.zeros(1000000)
+    >>> slice1 = tensor[:1000]
+    >>> slice2 = tensor[:10] # slice1 and slice2 share the same storage
+    >>> torch.save([slice1, slice2], 'share.pt')
+    >>> loaded_1, loaded_2 = torch.load('share.pt')
+    >>> loaded_1[0]
+    tensor(0.)
+    >>> loaded_2[0]
+    tensor(0.)
+    >>> loaded_2[0] = 1
+    >>> loaded_1[0] # loaded tensors still share storage
+    tensor(1.)
+
+Note that saving storage instead of tensor itself indicates the serialized file size might not match tensor size.
+In the example above the whole `tensor`'s storage (of size 1000000) is serialized instead of only slices.
+When tensor is expanded from a smaller storage, serialized file size might be smaller than tensor size as well.
+
+::
+
+    >>> a = torch.zeros(4).expand(4, 4)
+    >>> a.size()
+    torch.Size([4, 4])
+    >>> a.storage() # All columns of `a` share the same storage
+     0.0
+     0.0
+     0.0
+     0.0
+    [torch.FloatStorage of size 4]
+    >>> torch.save(a, 'a.pt')  # Only 4 float numbers are serialized
+    >>> loaded = torch.load('a.pt')
+    >>> loaded.storage()  # All colums of `loaded` share the same storage
+     0.0
+     0.0
+     0.0
+     0.0
+    [torch.FloatStorage of size 4]
+
+If you prefer to save only the slices without preserving storage sharing, call :meth:`~torch.Tensor.clone` before saving::
```

Comment:
I'm not sure you want to spin this as: If you don't want to preserve sharing, do this. But more if this causes issues like you're saved file contain a lot of unwanted data, you can use clone to fix that but note that this will break the sharing since you clone.
What do you think?

Reply:
SGTM!